mmcv/ops/csrc/common/cuda/assign_score_withk_cuda_kernel.cuh

```diff
@@ -22,34 +22,36 @@ __global__ void assign_score_withk_forward_cuda_kernel(
     const int O, const int aggregate, const T* points, const T* centers,
     const T* scores, const int64_t* knn_idx, T* output) {
   // ----- parallel loop for B, N1, K and O ---------
-  long i = blockIdx.x * blockDim.x + threadIdx.x;
-  if (i >= B * N1 * K * O) return;
-  // ------- loop for M ----------
-  const int b = (int)(i / (O * N1 * K));
-  const int o = (int)(i % (O * N1 * K) / (N1 * K));
-  const int n = (int)(i % (N1 * K) / K);
-  const int k = (int)(i % K);
-  const int cn = (int)knn_idx[b * K * N1 + n * K +
-                              0];  // The first neighbor is the center point
-  const int kn = (int)knn_idx[b * K * N1 + n * K + k];
-  if (kn >= N0 ||
-      kn < 0) {  // if index overflows, it is out of the neighborhood range
-    return;
-  }
-  assert(b < B);
-  assert(kn < N0);
-  assert(cn < N0);
-  assert(o < O);
-  assert(n < N1);
-  const int out_idx = b * N1 * O * K + o * N1 * K + n * K + k;
-  T val = output[out_idx];
-  for (int m = 0; m < M; m++) {
-    val += points[b * N0 * M * O + kn * M * O + m * O + o] *
-               scores[b * N1 * K * M + n * K * M + k * M + m] -
-           centers[b * N0 * M * O + cn * M * O + m * O + o] *
-               scores[b * N1 * K * M + n * K * M + k * M + m];
+  CUDA_1D_KERNEL_LOOP(index, B * O * N1 * K) {
+    long i = index;
```

Comment:
you can just use `CUDA_1D_KERNEL_LOOP(i, B * O * N1 * K)`

Reply:
Done